### PR TITLE
Backoffice grouping navbar

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description

1. Refactored backoffice navigation to allow collapsible sub-items
2. Made changes to allow new routes with no redirects back to cases
3. Fixed width and height on the cases list
4. Added a redirect for invalid routes to a valid route (404 route)

image

### CODE REVIEW

File: apps/backoffice-v2/package.json [hunk 1]
```diff
@@ -86,7 +87,7 @@
     "i18next-http-backend": "^2.1.1",
     "leaflet": "^1.9.4",
     "libphonenumber-js": "^1.10.49",
-    "lucide-react": "^0.144.0",
+    "lucide-react": "^0.239.0",
```
Comment (Omri-Levy) [inline apps/backoffice-v2/package.json:90]: Minimum version for the `Goal` icon used for the new route.

File: apps/backoffice-v2/src/Router/Router.tsx [hunk 2]
```diff
@@ -17,8 +17,14 @@ import { UnauthenticatedLayout } from '@/domains/auth/components/Unauthenticated
...
 import { unauthenticatedLayoutLoader } from '@/domains/auth/components/UnauthenticatedLayout/UnauthenticatedLayout.loader';
 import { Document } from '@/pages/Document/Document.page';
+import { NotFoundRedirect } from '@/pages/NotFound/NotFound';
 
 const router = createBrowserRouter([
+  {
+    path: '/*',
+    element: <NotFoundRedirect />,
+    errorElement: <RouteError />,
+  },
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/Router/Router.tsx:27]: Redirects invalid routes to a valid route.

File: apps/backoffice-v2/src/pages/CaseManagement/CaseManagement.page.tsx [hunk 3]
```diff
@@ -1,3 +1,8 @@
 import { Outlet } from 'react-router-dom';
+import { useSelectEntityFilterOnMount } from '@/domains/entities/hooks/useSelectEntityFilterOnMount/useSelectEntityFilterOnMount';
 
-export const CaseManagement = () => <Outlet />;
+export const CaseManagement = () => {
+  useSelectEntityFilterOnMount();
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/CaseManagement/CaseManagement.page.tsx:5]: Allows navigating to new routes without being redirected back to cases.

File: apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx [hunk 4]
```diff
@@ -43,7 +43,7 @@ export const useDocumentLogic = () => {
         );
       }
 
-      broadcastChannel.postMessage({
+      void broadcastChannel.postMessage({
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx:46]: ESLint change.

File: apps/backoffice-v2/src/Router/Router.tsx (hunk 2 above)
Comment (tomer-shvadron) [reply apps/backoffice-v2/src/Router/Router.tsx:27]: Shouldn't it be last? Or it doesn't matter?

File: apps/backoffice-v2/src/common/components/organisms/Header/Header.tsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ import { BottomActions } from './Header.BottomActions';
  */
 export const Header: FunctionComponent = () => {
   return (
-    <header className={`flex flex-col bg-[#F9FAFC] p-4`}>
+    <header className={`flex flex-col bg-[#F4F6FD] p-4 px-3`}>
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/common/components/organisms/Header/Header.tsx]: I would change `p-4` to `py-4`, even if it works, I find it helps to have the results more expectable. WDYT?

File: apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  const individualsFilters = useMemo(
+    () => filters?.filter(({ entity }) => entity === 'individuals'),
+    [filters],
+  );
+  const businessesFilters = useMemo(
+    () => filters?.filter(({ entity }) => entity === 'businesses'),
+    [filters],
+  );
+  const navItems = [
+    {
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx:19]: I think the categories should be added only if needed, i.e. if `businessFilters?.length` and same for individuals. Maybe you should ask Nitzan what he thinks

File: apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        individualsFilters?.map(({ id, name }) => ({
+          filterId: id,
+          text: name,
+          href: `/en/case-management/entities?filterId=${id}`,
+          key: `nav-item-${id}`,
+        })) ?? [],
+      key: 'nav-item-individuals',
+    },
+    {
+      text: 'Transaction Monitoring',
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx:44]: Same point for Transaction Monitoring - I'm guessing not all our customers will have it

File: apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx (hunk 4 above)
Comment (tomer-shvadron) [reply apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx:46]: Stupid rule. Should we change it?

File: apps/backoffice-v2/src/Router/Router.tsx (hunk 2 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/Router/Router.tsx:27]: Does not matter both from what i see and read.

File: apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx (hunk 4 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx:46]: It was a rule that was recently voted in. It was brought up to our attention that this rule catches an important class of bugs and errors.

File: apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx (hunk 6 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx:19]: Added a "no items found" fallback.

File: apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx (hunk 7 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx:44]: We want to show this to all of our customers.

File: apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx (hunk 4 above)
Comment (tomer-shvadron) [reply apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx:46]: Just to educate myself - what does it catch? How does it help us?

**Review by tomer-shvadron — APPROVED** [review]

File: apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx (hunk 4 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/Document/hooks/useDocumentLogic/useDocumentLogic.tsx:46]: 1. Human error - just recently we caught a bug due to a missing `await`
2. Try/catch behaves differently when an async return has `await` as opposed to without `await`
3. In general this rule also promotes using try/catch and catch https://typescript-eslint.io/rules/no-floating-promises/

**Pull request merged**